core/templates/dev/head/components/promo/PromoBarDirective.ts

```diff
@@ -36,10 +36,19 @@ oppia.directive('promoBar', [
         function() {
           var ctrl = this;
           var isPromoDismissed = function() {
-            return !!angular.fromJson(sessionStorage.promoIsDismissed);
+            try {
```

Comment:
I don't think we should do it like this ... can we create a SessionStorageService or something (do we have a local storage service already?) and do the check there? I don't think you need try/except here, it's too broad. We should just check whether sessionStorageExists or not and return false if it doesn't exist.